src/test/mgr-unit.t.sol

```diff
@@ -488,10 +519,27 @@ contract TinlakeManagerUnitTest is DSTest, DSMath {
           migrate();
     }
 
-    // function testRecover(uint wad) public {
-    //      if (ceiling < wad) return; // amount has to be below ceiling
-    //     testSink(wad);
-    //     recover(wad, 1);
-    // }
+    function testFullRecover(uint128 wad) public {
+        if (ceiling < wad) return; // amount has to be below ceiling
+        testCull(wad);
+        recover(wad, 1);
+    }
+
+    function testPartialRecover(uint128 wad) public {
+        if (ceiling < wad) return; // amount has to be below ceiling
+        if (wad < 2) return; 
+        testCull(wad);
+        recover(wad/2, 1);
+    }
 
+    function testFailRecoverisGlad(uint128 wad) public {
+        recover(wad, 1);
+    }
+
+    function testRecoverGlobalSettlement(uint128 wad) public {
+        if (ceiling < wad) return; // amount has to be below ceiling
+        testCull(wad);
+        cage();
+        recover(wad, 1);
```

Comment:
maybe we can add here:
```
assertEq(dai.balanceOf(self), wad);
```

Reply:
the assertions are in the recover function